### ISSUE TITLE
Conditional content should alias the user facing format name

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#4411]
Take this document

````markdown
---
format: gfm
---

::: {.content-visible when-format="gfm"}
Should be shown
:::

Some other text
````

This will not work because our `gfm` format is build on `commonmark`. 
https://github.com/quarto-dev/quarto-cli/blob/8ad60b0cf6990c34afc20b3e04609496793406d3/src/format/markdown/format-markdown.ts#L46-L55

So this will work 
````markdown
---
format: gfm
---

::: {.content-visible when-format="commonmark"}
Should be shown
:::

Some other text

````

and our global alias too 
````markdown
---
format: gfm
---

::: {.content-visible when-format="markdown"}
Should be shown
:::

Some other text
````

We should probably correctly alias our format name, shouldn't we ?

**Comment by cderv** [comment]
I am sure it was working before as I was using in example doc, and it worked at the time
https://github.com/quarto-dev/quarto-examples/blob/1ace3d789017770db1c665c7bdb1a2259eac5b1a/tabsets/tabsets-from-r-chunks/README.qmd#L7-L9
https://github.com/quarto-dev/quarto-examples/blob/1ace3d789017770db1c665c7bdb1a2259eac5b1a/tabsets/tabsets-from-r-chunks/README.md?plain=1#L4-L5

@jjallaire I believe this could be a regression from this change https://github.com/quarto-dev/quarto-cli/commit/657a8a32a3109f5b6280aa33d7e8d22228554c47 as before `gfmFormat()` was using gfm. 

My example above is from a few days before this change

I found it this way - but the issue is really with what to pass to `when-format`. Hugo has the same issue 
````markdown
---
format: hugo
---

::: {.content-visible when-format="hugo"}
Should be shown
:::

Some other text
````

Or is using `when-format="markdown"` the only way now ?

**Comment by cscheid** [comment]
@dragonstyle is this related to the conversation we were having yesterday in the context of gfm vs. markdown in the mermaid handlers?

**Comment by dragonstyle** [comment]
Yep it is the same underlying issue - `to` is now returning common+extensions instead of gfm. The format now includes an `identifier` which includes its baseFormat (in this case `gfm`) - we should likely be using that base from when checking conditionality.

**Comment by cscheid** [comment]
I've assigned myself because I know where to fix it but I won't have time to do it today. 

@cderv if you want to give this a shot the format checking happens in src/resources/pandoc/datadir/_format.lua:isFormat. Maybe it's enough to add "hugo", "gfm", "commonmark", etc to the check in line 168, but maybe we need to communicate more about the format. More tests will be needed here.

**Comment by mcanouil** [comment]
Related in some aspects to <https://github.com/quarto-dev/quarto-cli/discussions/3215#discussioncomment-4080101>.

**Comment by cscheid** [comment]
I completely sympathize with @mcanouil's here: we should have a "actually I really do mean this specific format" option. One immediate problem is that any changes to `when-format` will break backwards compatibility in a way that's hard to track and warn users.

I think the right solution is for us to come up with a consistent naming scheme for what we actually mean by format. The issue is that what goes into a "format" is actually a pretty complicated story. Here's a (probably incomplete) set of issues at play:

- Pandoc's format _variants_, such as `html+emoji`.
- Convenience under wildcards. We really often want an `html` wildcard that catches "stuff that a web browser reads".  Examples of complications:
  - `html5` is technically different from `html`, and so is `revealjs`.
  - `epub` is an "html" "format".
- quarto's custom formats: `acm-pdf`
- "harmless" lies that quarto tells its users: a `pdf` format declaration in a .qmd is actually sent to `pandoc` as `latex`; although pandoc supports `pdf` as an output format, quarto takes over the latex->pdf rendering part of the process.

I actually am now very reluctant to change this for 1.3.

**Comment by mcanouil** [comment]
Thanks for the explanation, I kind of figured that out while answering #4436 where I discovered we can set `format: markdown_strict+pipe_tables`.

**Comment by jjallaire** [comment]
Yes, we really need these matchers to be "smart" about all of the considerations that @cscheid mentions. We could introduce something like `when-format-config="acm-pdf"` that matches the exact format value within YAML config but I think we want to keep the main mechanism flexible. Agreed we should leave this alone for v1.3.

**Comment by mcanouil** [comment]
Maybe to keep it close to the current implementation, simply add a parameter determining the type of matching `when-format-matching=<exact/wildcard>`.

**Comment by jjallaire** [comment]
Or maybe `exact=true` ?

**Comment by mcanouil** [comment]
Shorter is probably better 😅

**Comment by cderv** [comment]
We should also add support for `confluence`
